sympy/series/formal.py

```diff
@@ -972,8 +978,19 @@ def infinite(self):
 
         return self.ind + inf_sum
 
+    @property
+    def fps_sym(self):
+        """Returns whether the function contains symbolic terms"""
+        ind = self.ind
+        s = ind.free_symbols.difference({self.x})
+        if len(s) == S.One:
+            return True
+        return False
+
     def _get_pow_x(self, term):
         """Returns the power of x in a term."""
+        if self.fps_sym:
+            return S.One
```

Comment:
Why is this change needed? I am not very sure about this. This breaks the purpose of the function (behaves differently for a special case).

Reply:
Suppose, for
`fps(x**n*exp(x),x).truncate(4) == x**n + x**n*x + x**n*x**2/2 + x**n*x**3/6 + O(x**4, x)`

Take any term, for eg. `x**n*x`. If you do a `as_base_exp()` operation on the term, we will get `xterm = x**n*x, pow_x = 1`. The problem comes if `x**n` term comes. `pow_x` then is `n`, a symbol. To generalize it, I kept return value as `S.One` for all symbolic fps.

Reply:
Can we avoid sending in the ``x**n`` term? It's a constant with respect to the algorithm.

Reply:
But in `polynomial` function of fps, the line `for i, t in enumerate(self):` will catch it. Are you telling we start `xk` from `(k, 1, oo)` instead of `(k, 0, oo)`, to get rid of the independent term?

Reply:
Ideally, one would expect for the following thing to work.

```
>>> f = fps(sin(x), x)
>>> f._get_pow_x(x**(n+1))
>>> n + 1 # correct
>>> f._get_pow_x(x**n*x)
1 # wrong
```

Try using ``powsimp``. eg.

```
>>> powsimp(x**n*x, x)
x**(n+1)
```